modules/tensor_mechanics/include/materials/crystal_plasticity/CrystalPlasticityHCPDislocationSlipBeyerleinUpdate.h

```diff
@@ -0,0 +1,231 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "CrystalPlasticityStressUpdateBase.h"
+
+class CrystalPlasticityHCPDislocationSlipBeyerleinUpdate;
+
+/**
+ * CrystalPlasticityHCPDislocationSlipBeyerleinUpdate computes the dislocation
+ * forest evolution for the prismatic, pyramidal, and basal slip systems
+ */
+
+class CrystalPlasticityHCPDislocationSlipBeyerleinUpdate : public CrystalPlasticityStressUpdateBase
+{
+public:
+  static InputParameters validParams();
+
+  CrystalPlasticityHCPDislocationSlipBeyerleinUpdate(const InputParameters & parameters);
+
+protected:
+//  TODO: add in capability to read the resistance parameters from the file?
```

Comment:
better ass an issue on github than a comment here

Reply:
yeah yeah. will do so prior to removing the wip tag